modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/JaxRsResourceRegistryImpl.java

```diff
@@ -50,6 +50,11 @@ public Object getPropertyValue(String className, String propertyName) {
 		return object;
 	}
 
+	@Override
+	public String getResourceValue(String className) {
```

Comment:
Mmmmm, I'm worried about this method. From my point of view, it is too coupled with the navigation menu implementation as you only want the class name.

Maybe a possible solution could be adding the properties like the actual implementation, but doing it with the DTO name you are providing.

Something like
```
@Override
	public Object getPropertyValueByDTO(String DTOclassName, String propertyName) {
		Object object = null;

		Map<String, Object> properties = _jaxRsModelProperties.get(
			className);

		if (properties != null) {
			object = properties.get(propertyName);
		}

		return object;
	}
```

Of course, it is necessary to change the `_jaxRsModelProperties` type and how we add this in the  ServiceTracker.

Why I am suggesting this? Because I saw a property that marks the JAX-RS application. 
The property `osgi.jaxrs.application.select` returns `(osgi.jaxrs.name=Liferay.Headless.Admin.User)`
and we can use that result to get the basePath instead of hardcode `headless-delivery`

If we can infer the basePath, we can get the contentURL for Category and Vocabulary and cover more navigation menu types.

Of course, all of these are just my thoughts, we can discuss it because could be a better approach or this approach is not a good one ✌️